File: backend/src/main/java/com/gfnqueuechecker/backend/entity/Account.java
package com.gfnqueuechecker.backend.entity;

import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "account")
@NoArgsConstructor
@Data
@AllArgsConstructor
public class Account {

    @Id
    @Column(name = "account_id")
    @SequenceGenerator(name = "account_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "account_seq")
    @NotNull
    private Long id;

    @Column(name = "username")
    private String username;

    @Column(name = "password")
    private String password;

    @Column(name = "role")
    private Long role;

}

File: frontend/src/app/model/Account.ts
export class Account {
  id: number;
  username: string;
  password: string;
  role: number;

  constructor(id: number, username: string, password: string, role: number) {
    this.id = id;
    this.username = username;
    this.password = password;
    this.role = role;
  }

}
